Godeps/_workspace/src/k8s.io/kubernetes/pkg/proxy/userspace/roundrobin.go

```diff
@@ -112,6 +112,17 @@ func isSessionAffinity(affinity *affinityPolicy) bool {
 	return true
 }
 
+// ServiceHasEndpoints checks whether a service entry has endpoints.
+func (lb *LoadBalancerRR) ServiceHasEndpoints(svcPort proxy.ServicePortName) bool {
+	lb.lock.Lock()
+	defer lb.lock.Unlock()
+	if state, exists := lb.services[svcPort]; exists && state != nil && len(state.endpoints) > 0 {
+		return true
+	} else {
```

Comment:
and also this shouldn't be an if - `state, exists := ....; return state != nil && ...`.